src/drivers/byt-ssp.c

```diff
@@ -457,16 +457,31 @@ static inline int ssp_set_loopback_mode(struct dai *dai, uint32_t lbm)
 {
 	struct ssp_pdata *ssp = dai_get_drvdata(dai);
 
-	trace_ssp("loo");
-	spin_lock(&ssp->lock);
+	trace_ssp("los");
 
-	ssp_update_bits(dai, SSCR1, SSCR1_LBM, lbm ? SSCR1_LBM : 0);
+	if (ssp->lbm == lbm)
+		return 0;
 
+	spin_lock(&ssp->lock);
+	ssp->lbm = lbm;
+	ssp_update_bits(dai, SSCR1, SSCR1_LBM, lbm ? SSCR1_LBM : 0);
 	spin_unlock(&ssp->lock);
 
 	return 0;
 }
 
+static inline int ssp_get_loopback_mode(struct dai *dai)
+{
+	struct ssp_pdata *ssp = dai_get_drvdata(dai);
+	int ret;
+
+	trace_ssp("log");
+	spin_lock(&ssp->lock);
+	ret = ssp->lbm;
```

Comment:
Do we really need this variable to track state ? cant we just use the SSP LBM bit for this ?

Reply:
ok, will use the register value directly,